fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/source/reader/FlinkSourceSplitReader.java

```diff
@@ -241,18 +251,33 @@ private void subscribeLog(SourceSplitBase split, long startingOffset) {
         } else {
             Long partitionId = tableBucket.getPartitionId();
             int bucket = tableBucket.getBucket();
-            if (partitionId != null) {
-                logScanner.subscribe(partitionId, bucket, startingOffset);
-            } else {
-                logScanner.subscribe(bucket, startingOffset);
+            try {
+                if (partitionId != null) {
+                    // Try to subscribe using the partition id.
+                    logScanner.subscribe(partitionId, bucket, startingOffset);
+                } else {
+                    // If no partition id, subscribe by bucket only.
+                    logScanner.subscribe(bucket, startingOffset);
+                }
+            } catch (Exception e) {
+                // Traverse the exception chain to check for PartitionNotExistException.
+                boolean partitionNotExist =
+                        ExceptionUtils.findThrowable(e, PartitionNotExistException.class)
+                                .isPresent();
+                if (partitionNotExist) {
+                    LOG.warn(
```

Comment:
We also should mark split as removed since the partition is removed and so that flink enumerator can remove this split state.